deployments/main.tf

```diff
@@ -31,19 +31,46 @@ module "dpe-sandbox-spacelift-development" {
   k8s_stack_deployments_name         = "DPE DEV Kubernetes Deployments"
   k8s_stack_deployments_project_root = "deployments/stacks/dpe-k8s-deployments"
 
+  auth0_stack_name         = "DPE DEV Auth0"
+  auth0_stack_project_root = "deployments/stacks/dpe-auth0"
+  auth0_domain             = "dev-57n3awu5je6q653y.us.auth0.com"
+  auth0_clients = [
+    {
+      name        = "bfauble - automation"
+      description = "App for testing signoz"
+      app_type    = "non_interactive"
+    },
+    {
+      name        = "schematic - Github Actions"
+      description = "Client for Github Actions to export telemetry data"
+      app_type    = "non_interactive"
+    },
+    {
+      name        = "schematic - Dev"
+      description = "Client for schematic deployed to AWS DEV to export telemetry data"
+      app_type    = "non_interactive"
+    },
+  ]
+
   aws_account_id = "631692904429"
   region         = "us-east-1"
 
   cluster_name = "dpe-k8-sandbox"
   vpc_name     = "dpe-sandbox"
 
-  vpc_cidr_block                         = "10.52.16.0/20"
-  public_subnet_cidrs                    = ["10.52.16.0/24", "10.52.17.0/24"]
+  vpc_cidr_block = "10.52.16.0/20"
+  # A public subnet is required for each AZ in which the worker nodes are deployed
+  public_subnet_cidrs                    = ["10.52.16.0/24", "10.52.17.0/24", "10.52.19.0/24"]
   private_subnet_cidrs_eks_control_plane = ["10.52.18.0/28", "10.52.18.16/28"]
   azs_eks_control_plane                  = ["us-east-1a", "us-east-1b"]
 
-  private_subnet_cidrs_eks_worker_nodes = ["10.52.20.0/22", "10.52.24.0/22", "10.52.28.0/22"]
-  azs_eks_worker_nodes                  = ["us-east-1a", "us-east-1b", "us-east-1c"]
+  private_subnet_cidrs_eks_worker_nodes = ["10.52.28.0/22", "10.52.24.0/22", "10.52.20.0/22"]
+  azs_eks_worker_nodes                  = ["us-east-1c", "us-east-1b", "us-east-1a"]
+
+  enable_cluster_ingress = true
+  enable_otel_ingress    = true
+  ssl_hostname           = "a09a38cc5a8d6497ea69c6bf6318701b-1974793757.us-east-1.elb.amazonaws.com"
+  auth0_jwks_uri         = "https://dev-57n3awu5je6q653y.us.auth0.com/.well-known/jwks.json"
 }
```

Comment:
The juggling of this is an unfortunate side affect of how the VPC module creates subnets in specific AZs. By doing this it allows us to:

1.  Create 3 public subnets in each AZ
2.  Create 2 private subnets in a, and b for the eks control plane
3.  Create 3 private subnets in each AZ for the private worker nodes.

Why this is needed:

1.  In order for the AWS load balancer to forward traffic to a private subnet, there needs to be a public subnet in that AZ.
2.  Worker nodes can be deployed to all 3 AZs, and because of this when a pod was scheduled into the c subnet, the load balancer could not forward traffic to that instance.